### PR TITLE
XYZ-32: Increases size of position to 128 characters.

### PR DESCRIPTION
**Pull request by mkraft**
#### Summary
Increases size of position to 128 characters.

#### Ticket Link
[XYZ-32](https://mattermost.atlassian.net/browse/XYZ-32)

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Added or updated unit tests
- [x] Has UI changes
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/mattermost-server/blob/master/i18n/en.json) updates

### CODE REVIEW

File: i18n/en.json [hunk 1]
```diff
@@ -5404,7 +5404,7 @@
   },
   {
     "id": "model.user.is_valid.position.app_error",
-    "translation": "Invalid position: must not be longer than 35 characters."
+    "translation": "Invalid position: must not be longer than 128 characters."
```
Comment (mkraft) [inline i18n/en.json]: @jasonblais Currently in production the validation message returns "Invalid position: must not be longer than 35 characters." However, it's actually allowing 64 characters (if you, say, make a request with curl). Do we want to keep the validation message consistent with the UI (soft) validation (ie 35) or change it to be consistent with the DB limit (ie 128)? I have no preference I just want to make sure there's awareness of this difference.

Comment (jasonblais) [reply i18n/en.json]: @mkraft Is this same error message used in the UI?

**Comment by jwilander** [comment]
gofmt is complaining:
```bash
--- /home/ubuntu/jenkins/workspace/mattermost-platform-pr-split/src/github.com/mattermost/mattermost-server/store/sqlstore/upgrade.go.orig	2018-01-24 20:49:53.766284114 +0000
+++ /home/ubuntu/jenkins/workspace/mattermost-platform-pr-split/src/github.com/mattermost/mattermost-server/store/sqlstore/upgrade.go	2018-01-24 20:49:53.766284114 +0000
@@ -342,7 +342,7 @@
 
 func UpgradeDatabaseToVersion47(sqlStore SqlStore) {
 	// if shouldPerformUpgrade(sqlStore, VERSION_4_6_0, VERSION_4_7_0) {
-	 	sqlStore.AlterColumnTypeIfExists("Users", "Position", "varchar(128)", "varchar(128)")
+	sqlStore.AlterColumnTypeIfExists("Users", "Position", "varchar(128)", "varchar(128)")
 	// 	saveSchemaVersion(sqlStore, VERSION_4_7_0)
 	// }
 }
gofmt failure
```

File: i18n/en.json (hunk 1 above)
Comment (mkraft) [reply i18n/en.json]: @jasonblais I'm not sure if the UI shows the message. You would have to change the markup to allow the input to be able to see it in the UI.

**Comment by mkraft** [comment]
@jwilander Fixed. That's what I get for editing Go file via Github UI.

File: i18n/en.json (hunk 1 above)
Comment (jasonblais) [reply i18n/en.json]: @mkraft Oh, right. Because we automatically prevent users from typing more than 35 characters.

Should be fine then.

**Pull request merged**